apps/hash-graph/lib/graph/src/store/postgres/ontology/entity_type.rs

```diff
@@ -25,197 +25,183 @@ impl<C: AsClient> PostgresStore<C> {
+                            &entity_type_vertex_id,
+                            OntologyEdgeKind::ConstrainsLinksOn,
+                            referenced_entity_type_vertex_id.clone(),
+                        );
+
+                        subgraph.insert_vertex(&referenced_entity_type_vertex_id, referenced_entity_type);
+
+                        entity_type_queue.push((
+                            referenced_entity_type_vertex_id,
+                            GraphResolveDepths {
+                                constrains_links_on: OutgoingEdgeResolveDepth {
+                                    outgoing: graph_resolve_depths.constrains_links_on.outgoing
+                                        - 1,
+                                    ..graph_resolve_depths.constrains_links_on
                                 },
-                                ..current_resolve_depths
+                                ..graph_resolve_depths
                             },
-                            temporal_axes.clone(),
+                            temporal_axes.clone()
                         ));
                     }
                 }
 
-                if let Some(link_mappings) = link_mappings {
-                    for (link_type_url, destination_type_urls) in link_mappings {
-                        if current_resolve_depths.constrains_links_on.outgoing > 0 {
-                            let link_type_vertex_id = EntityTypeVertexId::from(link_type_url);
-
-                            subgraph.insert_edge(
-                                &entity_type_id,
-                                OntologyEdgeKind::ConstrainsLinksOn,
-                                link_type_vertex_id.clone(),
-                            );
-
-                            queue.push((
-                                link_type_vertex_id,
-                                GraphResolveDepths {
-                                    constrains_links_on: OutgoingEdgeResolveDepth {
-                                        outgoing: current_resolve_depths
-                                            .constrains_links_on
-                                            .outgoing
-                                            - 1,
-                                        ..current_resolve_depths.constrains_links_on
-                                    },
-                                    ..current_resolve_depths
+                if graph_resolve_depths
+                    .constrains_link_destinations_on
+                    .outgoing
+                    > 0
+                {
+                    for referenced_entity_type in <Self as Read<EntityTypeWithMetadata>>::read(
+                        self,
+                        &Filter::<EntityTypeWithMetadata>::for_ontology_edge_by_entity_type_vertex_id(
+                            &entity_type_vertex_id,
+                            OntologyEdgeKind::ConstrainsLinkDestinationsOn,
+                            true,
```

Comment:
Why is this reversed?

Reply:
Answered in https://github.com/hashintel/hash/pull/2223#discussion_r1144646196 and https://github.com/hashintel/hash/pull/2223#discussion_r1144652948